src/document/reducer.ts

```diff
@@ -226,69 +286,56 @@ export function baseReducer<T, A extends Action, L>(
     dispatch?: SignalDispatch,
     options: ReducerOptions = {},
 ) {
+    // if there's skip value
+    // 0:0 1:0 2:0 3:0 4:1(new action)
+    // garbagecollector(0:0 1:0 2:0 3:0 4:1)
+    // 0:0 1:0 2:0 4:1
+    // replayedDocument = replayOperation(0:0 1:0 2:0)
+    // newDocument = {...document, state: {...replayedDocument.state} }
+
+    // resutl = reducer(newDocument, 4:1)
+
     const { skip, ignoreSkipOperations = false, reuseHash = false } = options;
```

Comment:
by default, we should process skip operations, this is because if you try to add an new operation with a skip value, you have to recalculate the state using the new skip value (run garbage collector), so the action is calculated over the right state.

we only do this once, if there's an skip operation, we call garbage collector with the document operations and then we run replayOperations using the cleared history. 

https://github.com/powerhouse-inc/document-model/pull/66/files#diff-f64607cae0373ae69cc3b37cc63d0d0ab2927dce9e3d57c3569fdae5222b5a86R245-R263

`replayOperations` then calls the reducer operation by operation with `ignoreSkipOperations = true` (no need to run the garbage collector again)

https://github.com/powerhouse-inc/document-model/blob/4a48fa6c893855cd22db16c7a09ffc5f544bc404/src/document/utils/base.ts#L369